src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -115,9 +124,9 @@ public DistributedApplication Build()
     /// <inheritdoc />
     public IResourceBuilder<T> AddResource<T>(T resource) where T : IResource
     {
-        if (Resources.FirstOrDefault(r => r.Name == resource.Name) is { } existingResource)
+        if (Resources.FirstOrDefault(r => string.Equals(r.Name, resource.Name, StringComparison.OrdinalIgnoreCase)) is { } existingResource)
         {
-            throw new DistributedApplicationException($"Cannot add resource of type '{resource.GetType()}' with name '{resource.Name}' because resource of type '{existingResource.GetType()}' with that name already exists.");
+            throw new DistributedApplicationException($"Cannot add resource of type '{resource.GetType()}' with name '{resource.Name}' because resource of type '{existingResource.GetType()}' with that name already exists. Resource names are case insensitive.");
```

Comment:
```suggestion
            throw new DistributedApplicationException($"Cannot add resource of type '{resource.GetType()}' with name '{resource.Name}' because resource of type '{existingResource.GetType()}' with that name already exists. Resource names are case-insensitive.");
```

from https://en.wikipedia.org/wiki/Case_sensitivity